Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut

```diff
@@ -951,33 +951,66 @@ string function FillInServerModsLabel( array<RequiredModInfo> mods )
 
 
 void function OnServerSelected( var button )
+{
+	thread OnServerSelected_Threaded( button )
+}
+
+void function OnServerSelected_Threaded( var button )
 {
 	if ( NSIsRequestingServerList() || NSGetServerCount() == 0 || file.serverListRequestFailed )
 		return
 
 	ServerInfo server = file.focusedServer
-
 	file.lastSelectedServer = server
 
+	// Count mods that have been successfully downloaded
+	bool autoDownloadAllowed = GetConVarInt( "allow_mod_auto_download" ) == 1
+	int downloadedMods = 0;
+
 	foreach ( RequiredModInfo mod in server.requiredMods )
 	{
 		if ( !NSGetModNames().contains( mod.name ) )
 		{
-			DialogData dialogData
-			dialogData.header = "#ERROR"
-			dialogData.message = format( "Missing mod \"%s\" v%s", mod.name, mod.version )
-			dialogData.image = $"ui/menu/common/dialog_error"
+			// Check if mod can be auto-downloaded
+			bool modIsVerified = NSIsModDownloadable( mod.name, mod.version )
 
-			#if PC_PROG
-				AddDialogButton( dialogData, "#DISMISS" )
+			// Display an error message if not
+			if (!modIsVerified || !autoDownloadAllowed)
+			{
+				DialogData dialogData
+				dialogData.header = "#ERROR"
+				dialogData.message = Localize ("#MISSING_MOD", mod.name, mod.version )
+				dialogData.image = $"ui/menu/common/dialog_error"
 
-				AddDialogFooter( dialogData, "#A_BUTTON_SELECT" )
-			#endif // PC_PROG
-			AddDialogFooter( dialogData, "#B_BUTTON_DISMISS_RUI" )
+				// Specify error (only if autoDownloadAllowed is set)
+				if (autoDownloadAllowed)
+				{
+					dialogData.message += "\n" + Localize( "#MOD_NOT_VERIFIED" )
+				}
 
-			OpenDialog( dialogData )
+				#if PC_PROG
+					AddDialogButton( dialogData, "#DISMISS" )
 
-			return
+					AddDialogFooter( dialogData, "#A_BUTTON_SELECT" )
+				#endif // PC_PROG
+				AddDialogFooter( dialogData, "#B_BUTTON_DISMISS_RUI" )
+
+				OpenDialog( dialogData )
+
+				return
+			}
+
+			// Launch download
+			else
```

Comment:
```suggestion
			}
			else // Launch download
```

this one is kinda personal preference tbh

Reply:
Addressed in 851d5853389abaabf92a2f0402b1f815e09faddd